.github/workflows/release.yml

```diff
@@ -6,32 +7,57 @@ on:
 
 jobs:
   release:
-    name: Release
+    name: 📦 Update version and release
     runs-on: ubuntu-latest
+    permissions:
+      contents: write
     steps:
-      - name: Checkout code
+      - name: ⤵️ Checkout code
         uses: actions/checkout@v6
+        with:
+          ref: main
+          token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Shouldn't this step checkout the tag of the the release and not main?

edit: ah, it's because of the commit and push in a later step. This should be fine since the releases always are made from the top of the main branch. _But_ the version bump will be not included in the release tag.

Reply:
Yeah I need to double check. The version is handled differently in custom integration than with a pypi lib. Need to update before release